core-common/src/main/java/org/glassfish/jersey/message/internal/InboundMessageContext.java

```diff
@@ -653,7 +653,9 @@ public URI apply(String value) {
         try {
             Set<Link> result = new HashSet<Link>(links.size());
             for (String l : links) {
-                result.add(Link.valueOf(l));
+                for (String p : l.split(",")) {
```

Comment:
I am not sure that splitting the value is enough. URI itself can contain ``,`` which means that this code would produce invalid result.